linkatos/firebase.py

```diff
@@ -12,21 +12,36 @@ def initialise(api_key, project_name):
     return pyrebase.initialize_app(config)
 
 
-def store_url(is_yes, url, user, password, firebase):
-    # do nothing if it's unnecessary
-    if not is_yes:
-        return False
-
-    # creates token every time maybe worth doing it once every 30m as they
-    # expire every hour
-    auth = firebase.auth()
-    user = auth.sign_in_with_email_and_password(user, password)
-
-    db = firebase.database()
-    data = {
-        "url": url
-    }
+def authenticate_user(username, password, auth):
+    user = auth.sign_in_with_email_and_password(username, password)
+    return user
+
+
+def get_db(firebase):
+    return firebase.database()
+
+
+def get_token(user):
+    return user['idToken']
+
+
+def store_url(url, db, token):
+    return db.child("users").push(to_data(url), token)
+
+
+def get_auth(firebase):
+    return firebase.auth()
+
+
+def to_data(url):
+    return {"url": url}
 
-    db.child("users").push(data, user['idToken'])
 
-    return False
+def connect_to_fb_and_store_url(url, username, password, firebase):
+    # the function should only be called if we need to store the url
+    auth = get_auth(firebase)
+    user = authenticate_user(username, password, auth)
+    token = get_token(user)
```

Comment:
I think, the unit you want is `token` or `get_token` if you like the verb. Or `user_token`.

At this level, I don't want to see the details for obtainint the token:

```py
credentials = {"username": username, "password": password} # I'd probably compose this outside this function as you want to treat them as a data unit.
token = user_token(firebase, credentials)
```